libdnf/module/ModulePackageContainer.cpp

```diff
@@ -1098,6 +1100,63 @@ void ModulePackageContainer::rollback()
     pImpl->persistor->rollback();
 }
 
+
+
+static TransactionItemAction
+moduleStateToTransactionItemAction(ModulePackageContainer::ModuleState state, bool in) {
+    switch (state) {
+        case ModulePackageContainer::ModuleState::ENABLED:
+            return in ? TransactionItemAction::ENABLE : TransactionItemAction::ENABLE_OUT;
+        case ModulePackageContainer::ModuleState::DISABLED:
+            return in ? TransactionItemAction::DISABLE : TransactionItemAction::DISABLE_OUT;
+        case ModulePackageContainer::ModuleState::UNKNOWN:
+            return in ? TransactionItemAction::RESET : TransactionItemAction::RESET_OUT;
+        case ModulePackageContainer::ModuleState::DEFAULT:
+            return in ? TransactionItemAction::RESET : TransactionItemAction::RESET_OUT;
+    }
+}
+
+
+/*
+ * Store changes in module stream states in history database.
+ */
+bool ModulePackageContainer::saveHistory(Swdb & swdb)
+{
+    bool result{false};
+    auto conn = swdb.getConn();
+
+    for (const auto & it : getChangedStreams()) {
+        result = true;
+        std::string name;
+        std::string oldStream;
+        ModuleState oldState;
+        std::string newStream;
+        ModuleState newState;
+        std::tie(name, oldStream, oldState, newStream, newState) = it;
+
+        auto new_stream = std::make_shared< ModuleStreamItem >(conn);
+        new_stream->setName(name);
+        new_stream->setStream(newStream);
+        // TODO: TransactionItemReason::DEPENDENCY
```

Comment:
If you don't plan on adding support for `DEPENDENCY` now, can you elaborate in the comment on what's missing and what's the plan?